### ISSUE TITLE
`enforce-consistent-variant-order` false positive report and autofix

### ISSUE THREAD

**Issue by aguo-ntech** [schoero/eslint-plugin-readable-tailwind#363]
### Description

The `enforce-consistent-variant-order` complains about this class and tries to autofix it to a class with a different meaning.

### Flavor

JSX

### Code Input

```tsx
*:first:appearance-auto
```

### Expected Behavior

```tsx
*:first:appearance-auto
```

### Actual Behavior

```tsx
first:*:appearance-auto
```

### ESLint Log

```shell
ESLint: Incorrect variant order. '*:first:appearance-auto' should be 'first:*:appearance-auto'. (better-tailwindcss/enforce-consistent-variant-order)
```

### ESLint Config

```typescript
Using the "tw4" config
```

### Versions

- Plugin: v4.4.0

**Comment by danciudev** [comment]
Same problem:

```js
variants: {
			variant: {
				default:
					"border-transparent bg-muted text-muted-foreground **:data-[slot=status-indicator]:bg-muted-foreground",
				error:
					"border-destructive/20 bg-destructive/10 text-destructive **:data-[slot=status-indicator]:bg-destructive",
				info: "border-blue-500/20 bg-blue-500/10 text-blue-600 **:data-[slot=status-indicator]:bg-blue-600 dark:text-blue-400 **:data-[slot=status-indicator]:dark:bg-blue-400",
				success:
					"border-green-500/20 bg-green-500/10 text-green-600 **:data-[slot=status-indicator]:bg-green-600 dark:text-green-400 **:data-[slot=status-indicator]:dark:bg-green-400",
				warning:
					"border-orange-500/20 bg-orange-500/10 text-orange-600 **:data-[slot=status-indicator]:bg-orange-600 dark:text-orange-400 **:data-[slot=status-indicator]:dark:bg-orange-400",
			},
		},
```

**Comment by tsyganov-vd** [comment]
Same here. 
My original classes:
``[&>*]:not-first:rounded-t-none [&>*]:not-last:rounded-b-none``

At first step, linter suggest simplify it: (The class: "[&>*]:not-first:rounded-t-none" can be simplified to "*:not-first:rounded-t-none")
``*:not-first:rounded-t-none *:not-last:rounded-b-none``
And its okay, everything is fine. Works correctly as my original.

But next step - ESLint: Incorrect variant order. ``*:not-first:rounded-t-none`` should be ``not-first:*:rounded-t-none``. (better-tailwindcss/enforce-consistent-variant-order)

But now this is incorrect:
``not-first:*:rounded-t-none`` is not the same as ``*:not-first:rounded-t-none``
